From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sat, 14 Oct 2023 13:42:58 -0500
Subject: [PATCH] Solus defaults

Changes:
- Default to vulkan over opengl for video driver
- Default to SDL as primary audio driver
- Default to VRR enabled when fullscreen
- Default to timing to audio buffer (seems to be the best way to prevent crackling)
- Default to dynamic rate control (also helps with crackling)
---
 gtk/src/gtk_config.cpp | 10 +++++-----
 gtk/src/gtk_sound.cpp  |  3 ++-
 snes9x.cpp             |  2 +-
 3 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/gtk/src/gtk_config.cpp b/gtk/src/gtk_config.cpp
index cd3bfd40..8a01ee30 100644
--- a/gtk/src/gtk_config.cpp
+++ b/gtk/src/gtk_config.cpp
@@ -90,11 +90,11 @@ int Snes9xConfig::load_defaults()
     rom_loaded = false;
     multithreading = false;
     splash_image = SPLASH_IMAGE_STARFIELD;
-    display_driver = "opengl";
+    display_driver = "vulkan";
     allow_opengl = false;
     allow_xv = false;
     allow_xrandr = false;
-    auto_vrr = false;
+    auto_vrr = true;
     osd_size = 24;
     force_inverted_byte_order = false;
     hires_effect = HIRES_NORMAL;
@@ -106,7 +106,7 @@ int Snes9xConfig::load_defaults()
     ui_visible = true;
     default_esc_behavior = 1;
     prevent_screensaver = false;
-    sound_driver = 0;
+    sound_driver = 0; //SDL
     sound_buffer_size = 48;
     sound_playback_rate = 7;
     sound_input_rate = 31950;
@@ -161,7 +161,7 @@ int Snes9xConfig::load_defaults()
     Settings.MultiPlayer5Master = true;
     Settings.UpAndDown = false;
     Settings.AutoSaveDelay = 0;
-    Settings.SkipFrames = 0;
+    Settings.SkipFrames = 2; // Sync to sound buffer
     Settings.Transparency = true;
     Settings.DisplayTime = false;
     Settings.DisplayFrameRate = false;
@@ -176,7 +176,7 @@ int Snes9xConfig::load_defaults()
     Settings.FrameTime = Settings.FrameTimeNTSC;
     Settings.BlockInvalidVRAMAccessMaster = true;
     Settings.SoundSync = false;
-    Settings.DynamicRateControl = false;
+    Settings.DynamicRateControl = true;
     Settings.DynamicRateLimit = 5;
     Settings.InterpolationMethod = DSP_INTERPOLATION_GAUSSIAN;
     Settings.HDMATimingHack = 100;
diff --git a/gtk/src/gtk_sound.cpp b/gtk/src/gtk_sound.cpp
index 157027d7..724105c6 100644
--- a/gtk/src/gtk_sound.cpp
+++ b/gtk/src/gtk_sound.cpp
@@ -58,6 +58,8 @@ std::vector<std::string> S9xGetSoundDriverNames()
 {
     std::vector<std::string> names;
 
+    // Make SDL the default
+    names.push_back("SDL");
 #ifdef USE_PORTAUDIO
     names.push_back("PortAudio");
 #endif
@@ -70,7 +72,6 @@ std::vector<std::string> S9xGetSoundDriverNames()
 #ifdef USE_PULSEAUDIO
     names.push_back("PulseAudio");
 #endif
-    names.push_back("SDL");
 
     return names;
 }
diff --git a/snes9x.cpp b/snes9x.cpp
index d4c757df..dc0a2254 100644
--- a/snes9x.cpp
+++ b/snes9x.cpp
@@ -229,7 +229,7 @@ void S9xLoadConfigFiles (char **argv, int argc)
 	Settings.SoundPlaybackRate          =  conf.GetUInt("Sound::Rate",                         48000);
 	Settings.SoundInputRate             =  conf.GetUInt("Sound::InputRate",                    31950);
 	Settings.Mute                       =  conf.GetBool("Sound::Mute",                         false);
-	Settings.DynamicRateControl         =  conf.GetBool("Sound::DynamicRateControl",           false);
+	Settings.DynamicRateControl         =  conf.GetBool("Sound::DynamicRateControl",           true);
 	Settings.DynamicRateLimit           =  conf.GetInt ("Sound::DynamicRateLimit",             5);
 	Settings.InterpolationMethod        =  conf.GetInt ("Sound::InterpolationMethod",          2);
 
